### ISSUE TITLE
CPU model setup

### ISSUE THREAD

**Issue by Demonsthere** [vagrant-libvirt/vagrant-libvirt#589]
Hi, my goal is to wrap a script that launches a virtual Raspberry Pi with the help of qemu inside of a vagrant box, with the libvirt as a provider. 
I have the vagrant box created from a Raspbian system, and a external kernel and initrd for the emulation purpose. 

my Vagrantfile is 
```ruby
# -*- mode: ruby -*-
# vi: set ft=ruby :
Vagrant.configure(2) do |config|
  # The most common configuration options are documented and commented below.
  # For a complete reference, please see the online documentation at
  # https://docs.vagrantup.com.

  config.vm.box = "arm"
  # vagrant issues #1673..fixes hang with configure_networks
  config.ssh.shell = "bash -c 'BASH_ENV=/etc/profile exec bash'"
  config.vm.provider :libvirt do |domain|
    domain.uri = 'qemu+unix:///system'
    domain.host = 'virtualized'
    domain.kernel = 'vmlinuz'
    domain.initrd = 'initrd'
    domain.cmd_line = 'root=/dev/mmcblk0p2 devtmpfs.mount=0 rw'
    domain.memory = 256
    domain.nested = true
    domain.driver = 'qemu'
    domain.machine_type = 'vexpress-a9'
    domain.machine_arch = 'arm'
    domain.cpu_mode = 'custom'
    domain.cpu_model = 'cortex-a9'
    domain.cpu_fallback = 'allow'
  end
end
```

And i get the error:
```bash
vagrant@default-debian8-local:~/data/vagrant$ vagrant up
Vagrant.require_plugin is deprecated and has no effect any longer.
Use `vagrant plugin` commands to manage plugins. This warning will
be removed in the next version of Vagrant.
Bringing machine 'default' up with 'libvirt' provider...
==> default: Starting domain.
There was an error talking to Libvirt. The error message is shown
below:

Call to virDomainCreateWithFlags failed: internal error: Unknown CPU model cortex-a9
```

However the cpu chosen is on the list of supported by qemu for that machine type
```
vagrant@default-debian8-local:~/data/vagrant$ qemu-system-arm -machine vexpress-a9 -cpu help
Available CPUs:
  arm1026
  arm1136
  arm1136-r2
  arm1176
  arm11mpcore
  arm926
  arm946
  cortex-a15
  cortex-a8
  cortex-a9
  cortex-m3
  pxa250
  pxa255
  pxa260
  pxa261
  pxa262
  pxa270-a0
  pxa270-a1
  pxa270
  pxa270-b0
  pxa270-b1
  pxa270-c0
  pxa270-c5
  sa1100
  sa1110
  ti925t
```

Is the libvirt provider capable of running those qemu CPU?

**Comment by infernix** [comment]
Try with `domain.machine_arch = 'armv7l'`, does that fix it?

**Comment by Demonsthere** [comment]
Hi,
Sadly no, neither changing that nor machine_type fixes the problem.
```ruby
domain.machine_type = 'virt'
domain.machine_arch = 'armv7l'
```

**Comment by infernix** [comment]
OK, here's a working config that works for me:

```ruby
ENV['VAGRANT_DEFAULT_PROVIDER'] = 'libvirt'

Vagrant.configure(2) do |config|
  # The most common configuration options are documented and commented below.
  # For a complete reference, please see the online documentation at
  # https://docs.vagrantup.com.

  config.vm.box = "arm"
  # vagrant issues #1673..fixes hang with configure_networks
  config.ssh.shell = "bash -c 'BASH_ENV=/etc/profile exec bash'"
  config.vm.provider :libvirt do |domain|
    domain.uri = 'qemu+unix:///system'
    domain.host = 'virtualized'
    domain.kernel = File.join(Dir.pwd, "vmlinuz")
    domain.initrd = File.join(Dir.pwd, "initrd")
    domain.cmd_line = 'root=/dev/mmcblk0p2 devtmpfs.mount=0 rw'
    domain.memory = 256
    domain.driver = 'qemu'
    domain.machine_type = 'vexpress-a9'
    domain.machine_arch = 'armv7l'
    domain.cpu_mode = 'custom'
    domain.cpu_model = 'cortex-a9'
    domain.cpu_fallback = 'allow'
    domain.graphics_type = 'none'
  end
end
```

This is with vagrant 1.8.1, vagrant-libvirt 0.0.33, libvirt 1.3.1 and qemu 2.5.

If that does not work for you with those versions, please post the output.

**Comment by psilar** [comment]
I had the same problem on a Debian Jessie amd64 machine trying to run an ARM version of Debian Wheezy with a similar configuration file.  I tried updating the version of vagrant but this didn't help.  Trying to run the created domain directly with virsh resulted in the same error so I'm guessing the problem is with my version of libvirt or qemu.  It would have been a headache to update the versions of those files so instead I found a simple workaround by changing two lines in my Vagrantfile:

```
      domain.cpu_mode = nil
      domain.cpu_model = nil
```

Once these settings were excluded from the XML, qemu-system-arm used the "Hypervisor Default" setting for the processor which worked fine for me.  Now I just have to figure out how to get the networking working :-).  Posting here in case anyone else finds this useful.
